test/e2e_node/split_disk_test.go

```diff
@@ -0,0 +1,338 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/kubernetes/pkg/features"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+	evictionapi "k8s.io/kubernetes/pkg/kubelet/eviction/api"
+	kubeletmetrics "k8s.io/kubernetes/pkg/kubelet/metrics"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	"k8s.io/kubernetes/test/e2e/nodefeature"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+// Eviction Policy is described here:
+// https://github.com/kubernetes/design-proposals-archive/blob/main/node/kubelet-eviction.md
+
+// InodeEviction tests that the node responds to node disk pressure by evicting only responsible pods.
+// Node disk pressure is induced by consuming all inodes on the node.
+
+// Existing Eviction tests are quite flaky and need some investigation
+// Flakiness arises from ordering
+// Stats is best effort and we evict based on stats being successful
+
+// Container runtime filesystem should display different stats for imagefs and nodefs
+
+var _ = SIGDescribe("Summary with split filesystem", framework.WithSlow(), framework.WithSerial(), framework.WithDisruptive(), nodefeature.SplitDisk, func() {
```

Comment:
does it only works on cri-o today? How will we filter it out for containerd? maybe this need a corres[ponding PR to test-infra to exclude this from containrd or skip in code

Reply:
maybe we skip similarly to how we skip on not windows https://github.com/kubernetes/kubernetes/blob/master/test/e2e/windows/utils.go#L89 ?

Reply:
I made a new beforeeach block that skips the tests if the feature gate is not enabled or they don't have split filesystem.